VideoRenderer/VideoRenderer/VideoStreamViewController.swift

```diff
@@ -154,40 +139,40 @@ public final class VideoStreamViewController: UIViewController, RendererProtocol
                 timeObserver = nil
 
                 currentPlayer = AVPlayer(url: props.content)
-                
-                var callbacks = PlayerObserver.Callbacks()
-                weak var this = self
-                
-                callbacks.failedState = {
-                    this?.dispatch?(.playbackFailed($0))
-                }
-                
-                callbacks.rateChanged = { new, old in
-                    guard new != old else { return }
-                    if new == 0 { this?.dispatch?(.playbackStopped) }
-                    else { this?.dispatch?(.playbackStarted) }
-                }
-                
-                callbacks.durationDefined = { duration, _ in
-                    this?.dispatch?(.durationReceived(duration))
-                }
-                
-                callbacks.endOfVideo = {
-                    this?.dispatch?(.playbackFinished)
-                }
-                
-                callbacks.playbackReady = {
-                    guard $0 == true else { return }
-                    this?.dispatch?(.playbackReady)
-                }
-                
-                callbacks.bufferedTimeUpdated = {
-                    this?.dispatch?(.bufferedTimeUpdated($0))
+
+                observer = SystemPlayerObserver(player: currentPlayer) { [weak self] event in
+                    switch event {
+                    case .didChangeItemStatus(_, let new):
+                        switch new {
+                        case .failed:
+                            let error: Error = {
+                                guard let error = currentPlayer.currentItem?.error else {
+                                    struct UnknownError: Error { let props: Renderer.Props }
+                                    return UnknownError(props: props)
+                                }
+                                return error
+                            }()
+                            self?.dispatch?(.playbackFailed(error))
+                        case .readyToPlay:
+                            self?.dispatch?(.playbackReady)
```

Comment:
Does it correct behavior to dispatch `playbackReady` here? At previous implementation this action triggered in case of `AVPlayerItem.isPlaybackBufferFull`or `AVPlayerItem.isPlaybackLikelyToKeepUp` became `true`